app/electron/preload.ts

```diff
@@ -0,0 +1,19 @@
+import { contextBridge, ipcRenderer } from 'electron';
+// Expose protected methods that allow the renderer process to use
+// the ipcRenderer without exposing the entire object
+contextBridge.exposeInMainWorld('desktopApi', {
+  send: (channel, data) => {
+    // allowed channels
+    const validChannels = ['updateAvailable', 'locale'];
```

Comment:
updateAvailable doesn't really belong to this patch though, as it's only used later.

Reply:
moved to the right place